StardewArchipelago/GameModifications/CodeInjections/Modded/JunimoShopInjections.cs

```diff
@@ -178,46 +179,87 @@ public static bool AnswerDialogueAction_Junimoshop_Prefix(GameLocation __instanc
+        }
+
+        private static void PurpleJunimoMoney(PurpleJunimo purpleJunimoOffer)
+        {
+            DaJunimo.setNewDialogue($"Oh very nice, have yellow rocks I found!  You like?");
+            Game1.drawDialogue(DaJunimo);
+            Game1.player.Money += 50000;
+            Game1.player.removeItemsFromInventory(purpleJunimoOffer.OfferedItem.Id, purpleJunimoOffer.Amount);
+        }
+
+        private static void PurpleJunimoDewdrop(PurpleJunimo purpleJunimoOffer)
+        {
+            var dewdropBerryId = _stardewItemManager.GetItemByName("Dewdrop Berry").Id;
+            var dewdropBerry = new StardewValley.Object(dewdropBerryId, 5);
+            Game1.player.holdUpItemThenMessage(dewdropBerry);
+            Game1.player.addItemByMenuIfNecessaryElseHoldUp(dewdropBerry);
+            DaJunimo.setNewDialogue($"Awh okay I was gonna sleep with it, but here you go!");
+            Game1.drawDialogue(DaJunimo);
+            Game1.player.removeItemsFromInventory(purpleJunimoOffer.OfferedItem.Id, purpleJunimoOffer.Amount);
+        }
+
+        private static void PurpleJunimoKiss(PurpleJunimo purpleJunimoOffer)
+        {
+            if (Game1.player.mailReceived.Contains("purpleJunimoKiss"))
                     {
                         DaJunimo.setNewDialogue($"You want me to kiss them TWICE?!  Wowie, but sorry!");
                         Game1.drawDialogue(DaJunimo);
-                        return false;
+                        return;
                     }
                     DaJunimo.setNewDialogue($"I will tonight!  Yay!");
                     Game1.drawDialogue(DaJunimo);
                     Game1.player.mailReceived.Add("purpleJunimoKiss");
-                    Game1.player.removeItemsFromInventory(checkOffer.OfferedItem.Id, checkOffer.Amount);
-                    return false;
-                }
+                    Game1.player.removeItemsFromInventory(purpleJunimoOffer.OfferedItem.Id, purpleJunimoOffer.Amount);
+        }
 
-                return false; // don't run original logic
+        // public void resetFriendshipsForNewDay()
+        public static void ResetFriendshipsForNewDay_KissForeheads_Postfix(Farmer __instance)
+        {
+            try
+            {
+                if (Game1.player.mailReceived.Contains("purpleJunimoKiss"))
+                {
+                    foreach (var friendship in Game1.player.friendshipData.Keys)
+                    {
+                        var friend = Game1.getCharacterFromName(friendship);
+                        var npc = friend ?? Game1.getCharacterFromName<Child>(friendship, false);
+                        if (npc == null)
+                        {
+                            continue;
+                        }
+                        Game1.player.changeFriendship(100, friend);
+                    }
+                    Game1.player.mailReceived.Remove("purpleJunimoKiss");
+                }
+                return; // don't run original logic
```

Comment:
you should not have these comments about running original logic on a postfix, as they don't apply. Just `return;`. The `return;` is sometimes not even necessary, but I often keep it around for ease of copy pasting when making prefixes out of postfixes, so that, I'm okay with